### ISSUE TITLE
Restart browser and reinstall extension when manifest changes

### ISSUE THREAD

**Issue by aklinker1** [aklinker1/wxt#10]

**Comment by aiktb** [comment]
I'm interested in working on this, but I've found the situation to be quite complex and would appreciate some guidance.

After reviewing the WXT source code, I found that the `detectDevChanges` function is used to determine the type of change and whether a full server restart is needed. However, I noticed that `generateManifest` is only called later during the rebuild process, after `detectDevChanges` has already returned. This means it's not possible to reload the extension immediately after `detectDevChanges`, since the new manifest hasn't been generated yet and reloading the extension at that point wouldn't make sense.

My questions are:

1. Is there a recommended way to handle manifest changes that occur outside of config file updates, given that the manifest is only available after the rebuild? 
2. For config file changes (such as the manifest key), a full server restart is triggered and the WebSocket connection is interrupted, so sending a reload extension command isn't possible. Maybe we can avoid full-restart when only the manifest key changes.
3. For other cases, I could compare the new manifest (from `generateManifest`) with the previous one after the rebuild, and reload the extension if they differ. However, this logic would be separate from the `changes.type` handling, which feels fragmented.

Do you have any suggestions or best practices for organizing this logic?

**Comment by PatrykKuniczak** [comment]
@aiktb Are you still eager to work on this?
If this is still affected, of course 😄

**Comment by aiktb** [comment]
> [@aiktb](https://github.com/aiktb) Are you still eager to work on this? If this is still affected, of course 😄

No, I've been very busy with work lately. Most importantly, I haven't figured out how to implement this yet.

**Comment by PatrykKuniczak** [comment]
@aiktb Are you still eager to take it?

**Comment by aiktb** [comment]
> [@aiktb](https://github.com/aiktb) Are you still eager to take it?

No
